scene/resources/shader.cpp

```diff
@@ -130,10 +130,16 @@ void Shader::get_shader_uniform_list(List<PropertyInfo> *p_params, bool p_get_gr
 	List<PropertyInfo> local;
 	RenderingServer::get_singleton()->get_shader_parameter_list(shader, &local);
 
-#ifdef TOOLS_ENABLED
+#if defined(TOOLS_ENABLED) && defined(MODULE_REGEX_ENABLED)
 	DocData::ClassDoc class_doc;
-	class_doc.name = get_path();
-	class_doc.is_script_doc = true;
+	bool generate_doc = EditorHelp::get_doc_data() != nullptr && Engine::get_singleton()->is_editor_hint() && !get_path().is_empty();
```

Comment:
This can be simplified if #95821 is merged. You should not need to check whether `get_doc_data() != nullptr`, the helper methods `EditorHelp::add_doc()/remove_doc()` etc can make those checks :)